airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -791,7 +808,6 @@ protected void verifyCheckpointStatesByRecords() throws Exception {
         .toListAndClose(secondBatchIterator);
     assertEquals(recordsToCreate, extractRecordMessages(dataFromSecondBatch).size());
     final List<AirbyteStateMessage> stateMessagesCDC = extractStateMessages(dataFromSecondBatch);
-    assertTrue(stateMessagesCDC.size() > 1, "Generated only the final state.");
```

Comment:
Shouldn't this be conditional on the postgres version as well?

Reply:
I wasn't 100% sure on the purpose of this line: On the comment above it says 

```    
// As first `read` operation is from snapshot, it would generate only one state message at the end
// of the process.
```
but it checks we will receive more than 1 state messages.

I'll probably adopt your approach to minimize the impact of this PR though.